records_mover/records/targets/table/move_from_dataframes_source.py

```diff
@@ -30,25 +30,27 @@ def __init__(self,
         super().__init__(prep, target_table_details, processing_instructions)
 
     def move(self) -> MoveResult:
-        if len(self.table_target.known_supported_records_formats()) != 0:
-            if self.table_target.can_move_from_fileobjs_source():
-                return self.move_from_dataframes_source_via_fileobjs()
-            else:
-                # Some databases, like Redshift, can't load from a
-                # stream, but can load from files on an object store
-                # they're pointed to.
-                return self.move_from_dataframes_source_via_records_directory()
+        target_supports_formats = len(self.table_target.known_supported_records_formats()) != 0
+        if (target_supports_formats and self.table_target.can_move_from_fileobjs_source()):
+            return self.move_from_dataframes_source_via_fileobjs()
+        elif (target_supports_formats and
+              self.table_target.can_move_from_temp_loc_after_filling_it()):
+            # Some databases, like Redshift, can't load from a
+            # stream, but can load from files on an object store
+            # they're pointed to.
+            return self.move_from_dataframes_source_via_temporary_records_directory()
```

Comment:
This is a slightly more esoteric usage of this interface, where the database table target has been told to take a dataframe and load the data inside.   There's a bit of a re-creation of a couple of parts of the move() algorithm here.  Not sure I'm happy with that re-creation; it might be an area for future refactoring if I can figure out how to make it work as part of the main move() algorithm.

But I digress.

This changes things so that when we are given dataframe (or create one as an intermediary step), we can ingest it into a database even if that database doesn't have a temporary bucket configured.